### PR TITLE
WIP: Meson build system

### PR DESCRIPTION
**Pull request by dcbaker**
I've had this kicking around for a while now, but I've never bothered to send it out because I could never figure out why some of the tests are failing. I still haven't figured it out, but I figured maybe if you guys cared about using meson you'd be willing to help me out :)

### CODE REVIEW

**Comment by dcbaker** [comment]
I was aware that Aaron had played with meson in the past, but he'd stopped.

I see a couple of differences between the two branches beyond me being based on current master and that branch being a couple back. I've used foreach loops to avoid a bunch of cut and paste, I don't have any automagic options, my arches are correct (meson will never return armv7l or ppc64le), I've avoided creating artificial dependency objects for wrapping a single shared library.

**Comment by fafryd1125** [comment]
When I asked Aaron about meson build and libglvnd, he said
>@aaronp24 said:.
As for merging this with upstream, that's up to @kbrenneman. When I asked him about it earlier, he said he didn't want to merge it until it works in Ubuntu 16.04. But the latest meson in xenial-backports is 0.40, and this branch requires 0.41 or higher. If we're going to

@dcbaker: Xserver won't start.
``` shell
[  8104.071] (II) Module glamoregl: vendor="X.Org Foundation"
[  8104.071]    compiled for 1.20.1, module version = 1.0.1
[  8104.071]    ABI class: X.Org ANSI C Emulation, version 0.4
[  8104.073] (EE)
[  8104.073] (EE) Backtrace:
[  8104.073] (EE) 0: /usr/bin/Xorg (xorg_backtrace+0x4f) [0x561f1629b24f]
[  8104.073] (EE) 1: /usr/bin/Xorg (0x561f16138000+0x166d0b) [0x561f1629ed0b]
[  8104.073] (EE) 2: /usr/lib/libc.so.6 (0x7ff4ec3f6000+0x37e00) [0x7ff4ec42de00]
[  8104.073] (EE) 3: /usr/lib/libEGL.so.1 (0x7ff4e05ac000+0x8254) [0x7ff4e05b4254]
[  8104.073] (EE) 4: /usr/lib/libEGL.so.1 (0x7ff4e05ac000+0x7eec) [0x7ff4e05b3eec]
[  8104.073] (EE) 5: /usr/lib/libpthread.so.0 (0x7ff4ec3d5000+0xf4ff) [0x7ff4ec3e44ff]
[  8104.073] (EE) 6: /usr/lib/libEGL.so.1 (0x7ff4e05ac000+0x7d1c) [0x7ff4e05b3d1c]
[  8104.073] (EE) 7: /usr/lib/libEGL.so.1 (eglQueryString+0xb0) [0x7ff4e05b12b0]
[  8104.073] (EE) 8: /usr/lib/libepoxy.so.0 (epoxy_has_egl_extension+0x18) [0x7ff4e0622028]
[  8104.073] (EE) 9: /usr/lib/xorg/modules/libglamoregl.so (glamor_egl_init+0x7a) [0x7ff4e072f6da]
[  8104.073] (EE) 10: /usr/lib/xorg/modules/drivers/amdgpu_drv.so (0x7ff4ed957000+0x1a067) [0x7ff4ed971067]
[  8104.073] (EE) 11: /usr/lib/xorg/modules/drivers/amdgpu_drv.so (0x7ff4ed957000+0xfbc8) [0x7ff4ed966bc8]
[  8104.073] (EE) 12: /usr/bin/Xorg (InitOutput+0xbcd) [0x561f162bd05d]
[  8104.073] (EE) 13: /usr/bin/Xorg (0x561f16138000+0x7cce5) [0x561f161b4ce5]
[  8104.074] (EE) 14: /usr/lib/libc.so.6 (__libc_start_main+0xf3) [0x7ff4ec41a223]
[  8104.074] (EE) 15: /usr/bin/Xorg (_start+0x2e) [0x561f1617312e]
[  8104.074] (EE)
[  8104.074] (EE) Segmentation fault at address 0x8
[  8104.074] (EE)
Fatal server error:
[  8104.074] (EE) Caught signal 11 (Segmentation fault). Server aborting
[  8104.074] (EE)
[  8104.074] (EE)
Please consult the The X.Org Foundation support

**Comment by kbrenneman** [comment]
For compatibility, I've been treating Ubuntu 16.04 and FreeBSD 11 as sort of a baseline minimum for what libglvnd needs to build on.

I wouldn't object to switching to Meson, but only if it can provide a sufficient degree of compatibility, and only if it's easier to maintain than the current autotools build. That means it has to be able to replace the autotools build without sacrificing compatibility -- if we have to maintain both, then it's just added effort and more chances for a mismatch with no benefit.

File: src/GL/meson.build [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+libGL = shared_library(
+  'GL',
+  ['libgl.c', g_libglxwrapper_c],
+  include_directories : [inc_include, inc_util, inc_gldispatch, inc_glx],
+  link_with : [libgldispatch, libglapi_gl, libutils_misc, libGLX],
```
Comment (fafryd1125) [inline src/GL/meson.build]: White space at the end of the line. :)

File: src/GLdispatch/meson.build [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+inc_gldispatch = include_directories('.', 'vnd-glapi')
+
+subdir('vnd-glapi')
+
+if use_tls
+  _linker_script = 'export_list_tls.ver'
+else
+  _linker_script = 'export_list_tld.ver'
+endif
+_linker_script = join_paths(meson.current_source_dir(), _linker_script)
```
Comment (fafryd1125) [inline src/GLdispatch/meson.build]: Should not it be like that?
`_version_script = join_paths(meson.current_source_dir(), _linker_script)`
`link_args : ['-Xlinker', '--version-script=@0@'.format(_version_script)],`

**Pull request closed without merging**